### PR TITLE
IOMMU/IPMMU updates

### PR DESCRIPTION
**Pull request by otyshchenko1**

### CODE REVIEW

**Comment by otyshchenko1** [comment]
Corresponding PR https://github.com/xen-troops/meta-xt-prod-devel/pull/311

**Comment by otyshchenko1** [comment]
Added two patches (to fix domain destroy):
Revert "xen/gnttab: fixed crash when grant mappings is used for DMA requests"
xen/domain: Call pci_release_devices() when releasing domain resources

File: xen/include/xen/pci.h [hunk 1]
```diff
@@ -132,13 +132,29 @@ struct pci_dev {
...
+#endif
 };
 
 #define for_each_pdev(domain, pdev) \
     list_for_each_entry(pdev, &(domain)->pdev_list, domain_list)
 
 #define has_arch_pdevs(d) (!list_empty(&(d)->pdev_list))
 
+#ifdef CONFIG_ARM
+static inline void pci_device_set_protected(struct pci_dev *pdev)
```
Comment (andr2000) [inline xen/include/xen/pci.h:147]: Is it possible that x86 can also implement something like this always returning 'true'?
So, we don't have ifdef ARM?

File: xen/drivers/passthrough/device_tree.c [hunk 2]
```diff
@@ -127,15 +127,38 @@ int iommu_release_dt_devices(struct domain *d)
...
 {
     const struct iommu_ops *ops = iommu_get_ops();
     struct dt_phandle_args iommu_spec;
     struct device *dev = dt_to_dev(np);
-    int rc = 1, index = 0;
+    int rc = NO_IOMMU, index = 0;
 
     if ( !iommu_enabled )
-        return 1;
+        return NO_IOMMU;
```
Comment (andr2000) [inline xen/drivers/passthrough/device_tree.c:304]: Are you going to return a local constant to the outside code?

**Comment by otyshchenko1** [comment]
pci_phase2_pre is about to be force-pushed with these patches in, so will close PR

**Pull request closed without merging**